git_machete/cmd.py

```diff
@@ -1570,15 +1573,174 @@ def flush_caches(self) -> None:
+
+        org: str
+        repo: str
+        _, (org, repo) = self.__derive_remote_and_github_org_and_repo()
+        current_user: Optional[str] = derive_current_user_login()
+        debug(f'create_github_pr({head})', f'organization is {org}, repository is {repo}')
+        debug(f'create_github_pr({head})', 'current GitHub user is ' + (current_user or '<none>'))
+
+        def slurp_file_or_empty(path: str) -> str:
+            try:
+                return open(path).read()
+            except IOError:
+                return ''
+
+        title: str = self.__git.popen_git("log", "-1", "--format=%s").strip()
+        description_path = self.__git.get_git_subpath('info', 'description')
+        description: str = slurp_file_or_empty(description_path)
+
+        ok_str = ' <green><b>OK</b></green>'
+        print(fmt(f'Creating a {"draft " if draft else ""}PR from `{head}` to `{base}`...'), end='', flush=True)
+        pr: GitHubPullRequest = create_pull_request(org, repo, head=head, base=base, title=title,
+                                                    description=description, draft=draft)
+        print(fmt(f'{ok_str}, see <b>{pr.html_url}</b>'))
+
+        milestone_path: str = self.__git.get_git_subpath('info', 'milestone')
+        milestone: str = slurp_file_or_empty(milestone_path).strip()
+        if milestone:
+            print(fmt(f'Setting milestone of PR #{pr.number} to {milestone}...'), end='', flush=True)
+            set_milestone_of_pull_request(org, repo, pr.number, milestone=milestone)
+            print(fmt(ok_str))
+
+        if current_user:
+            print(fmt(f'Adding `{current_user}` as assignee to PR #{pr.number}...'), end='', flush=True)
+            add_assignees_to_pull_request(org, repo, pr.number, [current_user])
+            print(fmt(ok_str))
+
+        reviewers_path = self.__git.get_git_subpath('info', 'reviewers')
+        reviewers: List[str] = utils.non_empty_lines(slurp_file_or_empty(reviewers_path))
+        if reviewers:
+            print(
+                fmt(f'Adding {", ".join(f"`{r}`" for r in reviewers)} as reviewer{"s" if len(reviewers) > 1 else ""} to PR #{pr.number}...'),
+                end='', flush=True)
+            add_reviewers_to_pull_request(org, repo, pr.number, reviewers)
+            print(fmt(ok_str))
+
+        self.__annotations[head] = f'PR #{pr.number}'
+        self.save_definition_file()
+
+    def __sync_before_creating_pr(self) -> None:
+
+        self.expect_at_least_one_managed_branch()
+        self.__empty_line_status = True
+
+        cb = self.__git.current_branch()
+        try:
+            self.expect_in_managed_branches(cb)
+        except MacheteException:
+            self.add(cb)
+
+        if self.__is_merged_to_upstream(cb):
+            return
+
+        s, remote = self.__git.get_strict_remote_sync_status(cb)
+        statuses_to_sync = (UNTRACKED, AHEAD_OF_REMOTE)
```

Comment:
Know what, you were right... I'm using this new features in my commercial project, and a force-push in case of `DIVERGED_FROM_AND_NEWER_THAN_REMOTE` would be clearly useful

Reply:
I'm not sure what to with `BEHIND_REMOTE` (`traverse` suggests `git pull`) or `DIVERGED_FROM_AND_OLDER_THAN_REMOTE` (`traverse` suggests `git reset --keep`)... probably displaying a warning to the user and proceeding with PR creation will be the most adequate action (?)